### PR TITLE
Add blog post about upcoming filesystem encryption and integrity features

### PR DESCRIPTION
**Pull request by Ashpan**
Blog post about filesystem encryption and integrity

Will need to change the date on the blog post but currently set to Dec 14 2021 so I could preview it on the github pages deployment

### CODE REVIEW

File: _posts/2021-12-14-upcoming-filesystem-encryption-integrity.adoc [hunk 1]
```diff
@@ -0,0 +1,83 @@
+---
+layout: post
+title: "Upcoming filesystem realm encryption and integrity support in Elytron"
+date: 2021-12-14
+tags:
+synopsis: An overview of the upcoming encryption and integrity support in Elytron Filesystem Security Realms.
+author: araskar
+---
+
+Two features that we have been working on for Elytron security is the ability to encrypt filesystem security realms with a SecretKey, and the ability to verify filesystem integrity with a KeyPair. This feature is planned to be released in WildFly 27.
```
[the remark this answers is not in the document]
Comment (Ashpan) [reply _posts/2021-12-14-upcoming-filesystem-encryption-integrity.adoc:10]: Fixed

File: _posts/2021-12-14-upcoming-filesystem-encryption-integrity.adoc [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        <password algorithm="bcrypt" format="base64">BQoQX9aqzoP3jgFXIWh9GOlIq1secNEbRofmD9lKHUd+/clh+dMwhusP</password>
+        <otp algorithm="otp-sha1" hash="UF2In5AIWEc=" seed="a2UxMjM0" sequence="500"/>
+    </credentials>
+    <attributes>
+        <attribute name="firstName" value="John"/>
+        <attribute name="lastName" value="Smith"/>
+        <attribute name="roles" value="Employee"/>
+        <attribute name="roles" value="Manager"/>
+        <attribute name="roles" value="Admin"/>
+    </attributes><Signature xmlns="http://www.w3.org/2000/09/xmldsig#"><SignedInfo><CanonicalizationMethod Algorithm="http://www.w3.org/TR/2001/REC-xml-c14n-20010315"/><SignatureMethod Algorithm="http://www.w3.org/2001/04/xmldsig-more#rsa-sha256"/><Reference URI=""><Transforms><Transform Algorithm="http://www.w3.org/2000/09/xmldsig#enveloped-signature"/></Transforms><DigestMethod Algorithm="http://www.w3.org/2001/04/xmlenc#sha256"/><DigestValue>Vt9Hea6ZOmYH9riBadwPNL2yY49B7X6S2OVfG4/1h4U=</DigestValue></Reference></SignedInfo><SignatureValue>Z+oXEAyowAMsnp6Myi2+8igRivY2LVeHy3OfxfAn/43v6hm4KLqb2rCFROMT9SrwowdIA0Ds9l0P&#13;
```
[the remark this answers is not in the document]
Comment (Ashpan) [reply _posts/2021-12-14-upcoming-filesystem-encryption-integrity.adoc:55]: Ah I missed that, fixed now

File: _posts/2021-12-14-upcoming-filesystem-encryption-integrity.adoc [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+Wazz1LsWFAi9NJJt3TEZ0PA8UdWUPeo5ddqpGPcjbpdOp1MgTyOqmYnJ3vXQYYsBGr+LCSfitATV&#13;
+mG8qbtz0DWneybWByh9osWRtjt/nd2HiFyKn/E0/kXY2eIuz/MO/LG+b28bTpX3WUR5/gJnNdxWe&#13;
+giVJmFk0vNf+cS5BQmMp3HUKl1hHihT+oTkUHBwGr0Ovbw33/8g97pqSf1itv0DBuL0TVg27hgq2&#13;
+KXWgdxmWsjEyTI6q2kjoY6ANopnXHR9b3AV5Tw==</Modulus><Exponent>AQAB</Exponent></RSAKeyValue></KeyValue></KeyInfo></Signature></identity>
+----
+
+== Filesystem Realm Encryption and Integrity Configuration
+To configure filesystem encryption, the follwing two attributes are required:
+
+* ``credential-store``: This attribute is a reference to the credential store that contains the SecretKey used to encrypt and decrypt the filesystem-realm.
```
[the remark this answers is not in the document]
Comment (Ashpan) [reply _posts/2021-12-14-upcoming-filesystem-encryption-integrity.adoc:69]: Fixed

File: _posts/2021-12-14-upcoming-filesystem-encryption-integrity.adoc [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+* ``secret-key``: This attribute is an alias to the secret key used to encrypt and decrypt the filesystem-realm. If the credential-store attribute is specified, this attribute is required.
```
[the remark this answers is not in the document]
Comment (Ashpan) [reply _posts/2021-12-14-upcoming-filesystem-encryption-integrity.adoc:71]: Fixed

File: _posts/2021-12-14-upcoming-filesystem-encryption-integrity.adoc [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+To configure filesystem integrity, another two attributes are required:
```
[the remark this answers is not in the document]
Comment (Ashpan) [reply _posts/2021-12-14-upcoming-filesystem-encryption-integrity.adoc:73]: Fixed

File: _posts/2021-12-14-upcoming-filesystem-encryption-integrity.adoc [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+* ``key-store``: This attribute is a reference to the KeyStore that contains the key pair used to sign and verify the signature of each identity.
```
[the remark this answers is not in the document]
Comment (Ashpan) [reply _posts/2021-12-14-upcoming-filesystem-encryption-integrity.adoc:75]: Fixed

File: _posts/2021-12-14-upcoming-filesystem-encryption-integrity.adoc [hunk 7]
```diff
@@ -0,0 +1,99 @@
+---
+layout: post
+title: "Upcoming filesystem realm encryption and integrity support in Elytron"
+date: 2021-12-14
+tags:
```
Comment (Skyllarr) [inline _posts/2021-12-14-upcoming-filesystem-encryption-integrity.adoc:5]: @Ashpan Just minor, but we could add filesystem-realm tag now, and maybe encryption as well

Comment (Ashpan) [reply _posts/2021-12-14-upcoming-filesystem-encryption-integrity.adoc]: Ah good catch, just added

**Pull request merged**